src/python_be.cc

```diff
@@ -886,6 +889,107 @@ ModelInstanceState::ProcessBLSCleanupRequest(
   }
 }
 
+void
+ModelInstanceState::ProcessMetricFamilyRequest(
+    const std::unique_ptr<IPCMessage>& message)
+{
+  AllocatedSharedMemory<CustomMetricsMessage> metrics_message =
+      Stub()->ShmPool()->Load<CustomMetricsMessage>(message->Args());
+  CustomMetricsMessage* metrics_message_ptr =
+      reinterpret_cast<CustomMetricsMessage*>(metrics_message.data_.get());
+  std::unique_ptr<PbString> pb_error_message;
+  PythonBackendException pb_exception(std::string{});
+  std::unique_ptr<MetricFamily> metric_family =
+      MetricFamily::LoadFromSharedMemory(
+          Stub()->ShmPool(), metrics_message_ptr->message);
+
+  ScopedDefer _([metrics_message_ptr] {
+    {
+      bi::scoped_lock<bi::interprocess_mutex> guard{metrics_message_ptr->mu};
+      metrics_message_ptr->waiting_on_stub = true;
+      metrics_message_ptr->cv.notify_all();
+      while (metrics_message_ptr->waiting_on_stub) {
+        metrics_message_ptr->cv.wait(guard);
+      }
+    }
+  });
+
+  try {
+    if (metric_family->RequestKind() == MetricFamilyNew) {
+      metrics_message_ptr->address =
+          metric_family->InitializeTritonMetricFamily();
+    } else if (metric_family->RequestKind() == MetricFamilyDelete) {
+      metric_family->ClearTritonMetricFamily();
+    } else {
+      pb_exception =
+          PythonBackendException("Unknown metric family request kind");
+    }
+  }
+  catch (const PythonBackendException& exception) {
+    pb_exception = exception;
+  }
+
+  if (pb_exception.what() != std::string{}) {
+    metrics_message_ptr->has_error = true;
+    LOG_IF_EXCEPTION(
+        pb_error_message =
+            PbString::Create(Stub()->ShmPool(), pb_exception.what()));
+    metrics_message_ptr->error = pb_error_message->ShmHandle();
+    metrics_message_ptr->is_error_set = true;
+  }
+}
+
+void
+ModelInstanceState::ProcessMetricRequest(
```

Comment:
It seems like there is a lot in common with `ProcessMetricRequest` and `ProcessMetricFamilyRequest`. Would it be possible to refactor the common part into a separate function?